### ISSUE TITLE
Additional Thing metadata

### ISSUE THREAD

**Issue by jtc42** [labthings/python-labthings#16]
Mozilla Web Things framework allows for additional descriptions of Things, such as device types, using IoT ontology. 

We should provide something similar. Ideally using an existing ontology....

**Comment by ChasNelson1990** [comment]
So, what you're looking for here is some sort of existing ontology for: all lab equipment? or just imaging? What level of detail do we need? Happy to search for one/if I can't find one try and build one from an existing data source, there are a few projects out there for automatically building ontologies from documents.

**Comment by ChasNelson1990** [comment]
Also, this relates to #8

**Comment by jtc42** [comment]
> So, what you're looking for here is some sort of existing ontology for: all lab equipment? or just imaging? What level of detail do we need? Happy to search for one/if I can't find one try and build one from an existing data source, there are a few projects out there for automatically building ontologies from documents.

So ideally I'd like some beautifully defined ontology for all scientific instrumentation. However, I realise this is completely infeasible. This is mostly an issue thread for discussion on sensible ways forward.

I think we need to be adopting whatever is closest to a common scientific ontology at some point, but I'm very open to suggestions.

Potentially useful starting point: [https://jcheminf.biomedcentral.com/articles/10.1186/s13321-016-0168-9](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-016-0168-9)

**Comment by ChasNelson1990** [comment]
So, SciData seems to be a level too abstract, right? It goes goes down to 'instrument' and all the granularity below that is lost?

And perhaps the OME-XML structure is a level less abstract than we need? And imaging specific... https://docs.openmicroscopy.org/ome-model/6.0.1/developers/model-overview.html

**Comment by jtc42** [comment]
AH yeah okay should have been clear that I want to avoid anything thats specific to imaging. One of the first targets I have for demo labs is an automatic photonics lab (rotating polarisers, translation stages etc).

It might be that an existing framework doesn't exist, because it'd be inconceivably broad, in which case rather than strictly define an ontology, it might be a better idea to instead give guidelines on how to describe components in a parsable way?

This is a tricky one though...

**Comment by ChasNelson1990** [comment]
Well....... if there's a corpus that we consider the key corpus for 'lab things' (no capitals) we could try automatic ontology extraction as a start... but I think you're right that it will be huge.

P.S. everything in your photonics lab I would also describe as imaging...

**Comment by jtc42** [comment]
> Well....... if there's a corpus that we consider the key corpus for 'lab things' (no capitals) we could try automatic ontology extraction as a start... but I think you're right that it will be huge.
> 
> P.S. everything in your photonics lab I would also describe as imaging...

I'll keep the issue open for feedback and we can just add suggestions as they come up.

Looking through the OME ontology, it hardly covers swathes of microscopy comprehensively, let alone photonics labs that don't obtain any images. I think you'd be hard pushed to describe counting photons as functions of polarisation optic positions using the OME ontology!

**Comment by ChasNelson1990** [comment]
True, I don't think the OME data schema has ever really been complete... which may have something to do with why it's never *really* taken off.

**Comment by ChasNelson1990** [comment]
I think perhaps focussing on clear guidelines would be the best solution.

**Comment by jtc42** [comment]
Also my suspicion why this is not a rabbit hole to commit to.... It's such an impossibly huge job that yeah, I think clear guidelines and community discussion might be best.

Guidelines for things like describing units of measurement for properties even would go a long way to helping. Something to think about, but I'll try not to lose sleep over...

**Comment by glyg** [comment]
Hi, I don't know if you are aware of the [NGFF initiative to re-define a file format standard for microscopy](https://github.com/ome/ngff), but there are very similar discussion going on there. That might also interest you regarding issue #13 to efficiently move are reference possibly big numerical data.
